### PR TITLE
Han Fan (Section 8)

### PR DESCRIPTION
**Pull request by hanfann**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment? 
1

Please fill in one error and/or error message you received while working on this assignment. 
Did not close my curly braces

What part of the assignment did you spend the most time on?
Figuring out how to use <br> line breaks

Comfort Level (1-5):
5

Completeness Level (1-5):
3

What did you think of this deliverable?
Simple but good for practise

Is there anything in this code that you feel pleased about?

### CODE REVIEW

**Comment by porkchop4** [comment]
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment?
3-4

Please fill in one error and/or error message you received while working on this assignment.
Mainly formatting errors or syntax errors

What part of the assignment did you spend the most time on?
creating a new function for the Korean version

Comfort Level (1-5):
3

Completeness Level (1-5):
3

What did you think of this deliverable?
Felt it was quite a step up from Part 1, but could have been because I did not attempt the more comfortable portion for Part 1

Is there anything in this code that you feel pleased about?

File: script.js [hunk 1]
```diff
@@ -1,4 +1,146 @@
+var attempts = 0;
+var wins = 0;
+var playerName = "";
+var gamemode = "normal";
+var currentWinner = "";
```
Comment (jerome-chua) [inline script.js:5]: Great setting up of your global values Han Fan

In future you may want to explore something know as SCREAM_CASE constants

I can touch more on this as the days go by, but it's simply another way to name variables that remain unchanged through out the program eg.
`
var NORMAL_GAME_MODE = "normal"
var KOREAN_MODE = "korean"
`

File: script.js [hunk 2]
```diff
@@ -1,4 +1,146 @@
...
+
+// right now attempts & win count carries over for normal and reversed game, but not korean version
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (playerName == "") {
+    playerName = input;
```
Comment (jerome-chua) [inline script.js:11]: I like how you intro the user to the game, well done!
Another possibility would have been to directly call
`playerName = input;` if that was what you were trying to achieve

File: script.js [hunk 3]
```diff
@@ -1,4 +1,146 @@
...
+
+  // generating computer choice
+  var genRandomNumber = Math.floor(Math.random() * 3 + 1);
+  var computerChoice = "scissors";
+  if (genRandomNumber == 1) {
+    computerChoice = "paper";
+  }
+  if (genRandomNumber == 2) {
+    computerChoice = "stone";
+  }
```
Comment (jerome-chua) [inline script.js:27]: To make this code reusable, could have put this in a function named something like 
`getComputerChoice()` and return either scissors/paper/stone as the return value for this function

File: script.js [hunk 4]
```diff
@@ -1,4 +1,146 @@
...
+    return "Game mode reversed. <br><br> Enter 'normal' anytime to revert back to the conventional game.";
+  }
+  if (input == "normal") {
+    gamemode = "normal";
+    return "Game mode is now normal";
+  }
+  if (input == "korean") {
+    gamemode = "korean";
+    return "Game mode is now Korean";
+  }
```
Comment (jerome-chua) [inline script.js:40]: Great job! Do explore the scream case variables mentioned earlier

File: script.js [hunk 5]
```diff
@@ -1,4 +1,146 @@
...
+    computerChoice = computerChoice + " ✂️";
+  } else if (computerChoice == "paper") {
+    computerChoice = computerChoice + " 🧻";
+  } else {
+    computerChoice = computerChoice + " 🪨";
+  }
+
+  if (input !== "scissors" && input !== "paper" && input !== "stone") {
+    return "Oops, please enter a valid choice (scissors / paper / stone)";
+  }
```
Comment (jerome-chua) [inline script.js:78]: Well done to cater to invalid cases!

File: script.js [hunk 6]
```diff
@@ -1,4 +1,146 @@
...
+      `Current Winner: ${currentWinner}`
+    );
+  } else {
+    if (currentWinner == "") {
+      return `Draw! You used ${input} and computer used ${computerChoice}`;
+    } else {
+      var koreanMessage =
+        "We have a winner! <br><br>" +
+        `You used ${input} and the computer used ${computerChoice}, so the ultimate winner is ${currentWinner}!!!`;
+      currentWinner = "";
```
Comment (jerome-chua) [inline script.js:141]: Were you trying to restart the game here, by setting winner name back to an empty string? 
A suggestion would be to encapsulate all the things that need to be reset in a function, so it'll be clearer what you are trying to say through your code

File: script.js [hunk 7]
```diff
@@ -1,4 +1,146 @@
...
+    myOutputValue = "You win!";
+  } else if (input == "paper" && computerChoice == "stone") {
+    wins += 1;
+    myOutputValue = "You win!";
+  } else if (input == "stone" && computerChoice == "scissors") {
+    wins += 1;
+    myOutputValue = "You win!";
+  } else if (computerChoice == input) {
+    myOutputValue = "DRAW!!";
+  } else myOutputValue = "You lose...";
```
Comment (jerome-chua) [inline script.js:111]: Great job nailing down all the appropriate conditions!

File: script.js [hunk 8]
```diff
@@ -1,4 +1,146 @@
...
+  console.log(currentWinner);
+  if (myOutputValue == "You win!") {
+    currentWinner = "you";
+    return (
+      "MUK-JJI-PPA!! <br><br>" +
+      `You used ${input} and computer used ${computerChoice}` +
+      "<br><br> Quickly enter another choice of either scissors, paper or stone! <br><br>" +
+      `Current Winner: ${currentWinner}`
+    );
+  }
```
Comment (jerome-chua) [inline script.js:125]: This is a unique way of determining the steps that follow after the winner has been found.

**Pull request closed without merging**